File: leetcode_go/question/_1401_1500/_1450/sol_1450.go
// Package _1450 在既定时间做作业的学生人数
// @author goinhn
// @data 2022/8/19
package _1450

// 遍历
// time:O(n)
// space:O(1)
func busyStudent(startTime []int, endTime []int, queryTime int) int {
	if startTime == nil || len(startTime) == 0 || endTime == nil || len(endTime) == 0 {
		return 0
	}

	count := 0
	for i := 0; i < len(startTime); i++ {
		if startTime[i] <= queryTime && endTime[i] >= queryTime {
			count++
		}
	}

	return count
}

File: leetcode_java/java/com/goinhn/thorwcode/leetcode/question/_1401_1500/Sol1450.java
package com.goinhn.thorwcode.leetcode.question._1401_1500;

/**
 * 1450.在既定时间做作业的学生人数
 *
 * @author goinhn
 * @date 2022/8/19
 */
public class Sol1450 {

    /**
     * 遍历
     * time:O(n)
     * space:O(1)
     */
    private static class Solution1 {

        public int busyStudent(int[] startTime, int[] endTime, int queryTime) {
            if (startTime == null || startTime.length == 0 || endTime == null || endTime.length == 0) {
                return 0;
            }

            int count = 0;
            for (int i = 0; i < startTime.length; i++) {
                if (startTime[i] <= queryTime && endTime[i] >= queryTime) {
                    count++;
                }
            }

            return count;
        }

    }

}
